src/webrtc/peerconnection.ts

```diff
@@ -317,47 +325,47 @@ export class PeerConnectionClass implements PeerConnection<signals.Message> {
     }
   }
 
-  private handleOfferSignalMessage_ =
-      (description:freedom_RTCPeerConnection.RTCSessionDescription) : void => {
-    this.breakOfferTie_(description)
-      .then(() => {
-        this.state_ = State.CONNECTING;
-        // initial offer from peer
-        return this.pc_.setRemoteDescription(description)
-      })
-      .then(this.pc_.createAnswer)
-      .then((d:freedom_RTCPeerConnection.RTCSessionDescription) => {
-        // As with the offer, we must emit the signal before
-        // setting the local description to ensure that we send the
-        // ANSWER before any ICE candidates.
-        this.emitSignalForPeer_({
-          type: signals.Type.ANSWER,
-          description: {
-            type: d.type,
-            sdp: d.sdp
-          }
-        });
-        this.pc_.setLocalDescription(d);
-      })
-      .catch((e) => {
-        this.closeWithError_('Failed to connect to offer:' +
-            e.toString());
+  private handleOfferSignalMessage_ = (
+      description:freedom_RTCPeerConnection.RTCSessionDescription)
+      : Promise<void> => {
+    return this.breakOfferTie_(description).then(() => {
+      this.state_ = State.CONNECTING;
+      // initial offer from peer
+      return this.pc_.setRemoteDescription(description)
+    }).then(this.pc_.createAnswer).then(
+        (d:freedom_RTCPeerConnection.RTCSessionDescription) => {
+      // As with the offer, we must emit the signal before
+      // setting the local description to ensure that we send the
+      // ANSWER before any ICE candidates.
+      this.emitSignalForPeer_({
+        type: signals.Type.ANSWER,
+        description: {
+          type: d.type,
+          sdp: d.sdp
+        }
       });
+      return this.pc_.setLocalDescription(d);
+    }).catch((e) => {
+      this.closeWithError_('Failed to connect to offer:' +
+          e.toString());
+    });
   }
 
-  private handleAnswerSignalMessage_ =
-      (description:freedom_RTCPeerConnection.RTCSessionDescription) : void => {
-    this.pc_.setRemoteDescription(description).catch((e) => {
+  private handleAnswerSignalMessage_ = (
+      description:freedom_RTCPeerConnection.RTCSessionDescription)
+      : Promise<void> => {
+    return this.pc_.setRemoteDescription(description).catch((e) => {
       this.closeWithError_('Failed to set remote description: ' +
         JSON.stringify(description) + '; Error: ' + e.toString());
     });
   }
 
   private handleCandidateSignalMessage_ = (
-      candidate:freedom_RTCPeerConnection.RTCIceCandidate) : void => {
+      candidate:freedom_RTCPeerConnection.RTCIceCandidate)
+      : Promise<void> => {
     // CONSIDER: Should we be passing/getting the SDP line index?
     // e.g. https://code.google.com/p/webrtc/source/browse/stable/samples/js/apprtc/js/main.js#331
-    this.pc_.addIceCandidate(candidate).catch((e: Error) => {
+    return this.pc_.addIceCandidate(candidate).catch((e: Error) => {
```

Comment:
No. I just figured since `addIceCandidate` *does* return a promise that it was no harm.